source-control/git-guidance/git-lfs-and-vfs.md

```diff
@@ -0,0 +1,156 @@
+* If not set up properly:
+  * Binary files committed through Git LFS are not visible as Git will only download the data describing the large file
+  * Committing large binaries will push the full binary to the repository
+* Git cannot merge the changes from two different versions of a binary file; file locking mitigates this
+* Azure Repos do not support using SSH for repositories with Git LFS tracked files - for more information see the Git LFS [authentication documentation](https://github.com/git-lfs/git-lfs/blob/master/docs/api/authentication.md)
+
+### Installation and use of Git LFS
+
+Go to [https://git-lfs.github.com](https://git-lfs.github.com) and download and install the setup from there.
+
+For every repository you want to use LFS, you have to go through these steps:
+
+* Setup LFS for the repo:
+
+```shell
+git lfs install
+```
+
+* Indicate which files have to be considered as large files (or binary files). As an example, to consider all Photoshop files to be large:
+
+```shell
+git lfs track "*.psd"
+```
+
+There are more fine grained ways to indicate files in a folder and more. See the [Git LFS Documentation](https://github.com/git-lfs/git-lfs/tree/master/docs?utm_source=gitlfs_site&utm_medium=docs_link&utm_campaign=gitlfs).
+
+With these commands a `.gitattribute` file is created which contains these settings and must be part of the repository.
+
+From here on you just use the standard git commands to work in the repository. The rest will be handled by Git and Git LFS.
+
+### Common LFS commands
+
+Install git lfs
+
+```bash
+git lfs install       # windows
+sudo apt-get git-lfs  # linux
+```
+
+See the [Git LFS installation instructions](https://github.com/git-lfs/git-lfs/wiki/Installation) for installation on other systems
+
+Track .mp4 files with Git LFS
+
+```bash
+git lfs track '*.mp4'
+```
+
+Update the `.gitattributes` file listing the files and patterns to track
+
+```bash
+*.mp4 filter=lfs diff=lfs merge=lfs -text
+docs/images/* filter=lfs diff=lfs merge=lfs -text
+```
+
+List all patterns tracked
+
+```bash
+git lfs track
+```
+
+List all files tracked
+
+```bash
+git lfs ls-files
+```
+
+Download files to your working directory
+
+```bash
+git lfs pull
+git lfs pull --include="path/to/file"
+```
+
+## VFS for Git
+
+Imagine a large repository with multiple projects in it, for instance for all kinds of features. A developer in a specific team is working on just one feature. He doesn't require *all* files of the repo to work on that one feature. But normally with Git, you will clone *all* files in the repo.
+
+VFS for Git (or Virtual File System for Git) solves this problem, as it will only download what you need to your local machine. But if you look in the file system, e.g. with Windows Explorer, it will show all the folders and files including the correct file sizes.
+
+When you use VFS for Git for a repository a process is started that does all the work to hide that downloading.
```

Comment:
```suggestion
```

Reply:
I don't think this sentence is needed - and it is a little confusing as it suggests that it is in fact downloading but the download is hidden to the user (or is that what really happens?)